### PR TITLE
Changing log_scalar step unit to batch # instead of epoch

### PR DESCRIPTION
**Pull request by corey-nm**
- Currently torchvision logs scalars with a step unit of epoch #. All other modifiers use batch # as unit for step, which causes charts not to be displayed in WANDB. This changes the step unit to match for modifiers.
- Fixing bug where using `--gradient-accum-steps` did not adjust `steps_per_epoch` passed to manager
- Also renaming `--print-freq` -> `--logging-steps` and properly using everywhere.

# Test Plan

Tested this command:
```bash
sparseml.image_classification.train --checkpoint-path zoo:cv/classification/efficientnet_v2-s/pytorch/sparseml/imagenet/base-none --arch-key efficientnet_v2_s --output-dir efficientnet-tmp/ --recipe dense_finetune.yaml --dataset-path /home/corey/.cache/nm_datasets/imagenette/imagenette-160/ --batch-size 4 --gradient-accum-steps 4 --auto-augment imagenet --random-erase 0.1 --label-smoothing 0.1 --mixup-alpha 0.2 --cutmix-alpha 1.0 --weight-decay 0.00002 --norm-weight-decay 0.0 --train-crop-size 300 --model-ema --val-crop-size 384 --val-resize-size 384 --ra-sampler --ra-reps 4 --logging-steps 100
```

with this recipe:
```yaml
version: 1.1.0

# General Variables
num_epochs: 3
lr_warmup_epochs: 1
init_lr: 0.0
warmup_lr: 1e-5
final_lr: 0.0
weight_decay: 0.00001

# Modifiers
training_modifiers:
  - !EpochRangeModifier
    start_epoch: 0
    end_epoch: eval(num_epochs)

  - !LearningRateFunctionModifier
    start_epoch: 0
    end_epoch: eval(lr_warmup_epochs)
    lr_func: linear
    init_lr: eval(init_lr)
    final_lr: eval(warmup_lr)

  - !LearningRateFunctionModifier
    start_epoch: eval(lr_warmup_epochs)
    end_epoch: eval(num_epochs)
    lr_func: cosine
    init_lr: eval(warmup_lr)
    final_lr: eval(final_lr)

  - !SetWeightDecayModifier
    start_epoch: 0
    weight_decay: eval(weight_decay)
```

# Wandb output

image

### CODE REVIEW

**Review by bfineran — DISMISSED** [review]
great PR description 👍

File: src/sparseml/pytorch/torchvision/train.py [hunk 1]
```diff
@@ -862,7 +884,7 @@ def new_func(*args, **kwargs):
     type=float,
     help="minimum lr of lr schedule",
 )
-@click.option("--print-freq", default=10, type=int, help="print frequency")
+@click.option("--logging-steps", default=10, type=int, help="print frequency")
```
[the remark this answers is not in the document]
Comment (rahul-tuli) [reply src/sparseml/pytorch/torchvision/train.py:887]: Also should we show a deprecation message if `print-freq` is used?

**Review by KSGulin — APPROVED** [review]
Nice!

**Review by rahul-tuli — DISMISSED** [review]
LGTM, specially like the importing `utils` directly instead of `from ...utils import (.....)`

File: src/sparseml/pytorch/torchvision/train.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (corey-nm) [reply src/sparseml/pytorch/torchvision/train.py:887]: Good callout, probably yes.

[the remark this answers is not in the document]
Comment (corey-nm) [reply src/sparseml/pytorch/torchvision/train.py]: Done!

**Review by bfineran — APPROVED** [review]

File: src/sparseml/pytorch/torchvision/train.py [hunk 2]
```diff
@@ -885,7 +890,13 @@ def new_func(*args, **kwargs):
     type=float,
     help="minimum lr of lr schedule",
 )
-@click.option("--logging-steps", default=10, type=int, help="print frequency")
+@click.option("--print-freq", default=None, type=int, help="DEPRECATED. Does nothing.")
```
Comment (rahul-tuli) [inline src/sparseml/pytorch/torchvision/train.py:893]: nice!

**Review by KSGulin — APPROVED** [review]

File: src/sparseml/pytorch/torchvision/train.py [hunk 3]
```diff
@@ -129,7 +134,7 @@ def train_one_epoch(
             batch_size / (time.time() - start_time)
         )
 
-        if i % args.logging_steps == 0:
+        if num_optim_steps % args.logging_steps == 0:
```
Comment (rahul-tuli) [inline src/sparseml/pytorch/torchvision/train.py:137]: nice!

**Review by rahul-tuli — APPROVED** [review]
